static/app/components/events/interfaces/performance/spanEvidenceKeyValueList.tsx

```diff
@@ -94,20 +94,30 @@ const NPlusOneDBQueriesSpanEvidence = ({
   event,
   parentSpan,
   offendingSpans,
-}: SpanEvidenceKeyValueListProps) => (
-  <PresortedKeyValueList
-    data={
-      [
-        makeTransactionNameRow(event),
-        parentSpan ? makeRow(t('Parent Span'), getSpanEvidenceValue(parentSpan)) : null,
-        makeRow(
-          t('Repeating Spans (%s)', offendingSpans.length),
-          getSpanEvidenceValue(offendingSpans[0])
-        ),
-      ].filter(Boolean) as KeyValueListData
-    }
-  />
-);
+}: SpanEvidenceKeyValueListProps) => {
+  const dbSpans = offendingSpans.filter(span => span.op === 'db');
+  const uniqueDbSpans = dbSpans.filter(span => {
+    return offendingSpans.find(s => s.description === span.description) === span;
+  });
+  const repeatingSpanRows = uniqueDbSpans.map((span, i) => {
+    return makeRow(
+      i === 0 ? t('Repeating Spans (%s)', dbSpans.length) : '',
+      getSpanEvidenceValue(span)
+    );
+  });
```

Comment:
I think it would be a bit cleaner to omit the return on the arrow functions (not needed if your returning right away), and just combining the filter and map.
```suggestion
  const repeatingSpanRows = dbSpans
    .filter(span => offendingSpans.find(s => s.description === span.description) === span)
    .map((span, i) =>
      makeRow(
        i === 0 ? t('Repeating Spans (%s)', dbSpans.length) : '',
        getSpanEvidenceValue(span)
      )
    );
```

Reply:
👍 Good call!